racket-cmd.el

```diff
@@ -77,39 +77,74 @@ that Racket Mode can find its own run.rkt file.")
+the rempote host and return the latter.
+
+If PATH is a local path then just retuirn
+`racket--rkt-source-dir'. If PATH is nil we assume we are
+operating locally and `racket--rkt-source-dir' is returned.
+
+The return value of this function is to be used as an argument to
+the racket repl invocation so it will have no tramp prefix. "
+  (if (or (null path) (not (file-remote-p path)))
+      racket--rkt-source-dir
+    (let ((remote-file-struct (tramp-dissect-file-name path)))
+      (setf (tramp-file-name-localname remote-file-struct)
+            racket--rkt-remote-source-dir)
+      (let ((remote-dir-name (tramp-make-tramp-file-name remote-file-struct)))
+        (unless (tramp-handle-file-directory-p remote-dir-name)
+          ;; COPY-DIRECTORY likes to create symlinks when the source
+          ;; is a symling (eg straight keeps package repos in a
+          ;; symlinked dir). This will never work when copying to a
+          ;; remote host. Fortunately we can replace every symlink
+          ;; create with a file copy.
+          (flet ((make-symbolic-link (src dst x) (copy-file src dst t nil)))
+           (copy-directory racket--rkt-source-dir remote-dir-name
+                           nil t t))))
+      racket--rkt-remote-source-dir)))
+
 (defun racket--cmd-open ()
   ;; Avoid multiple processes/buffers like "racket-process<1>".
   (racket--cmd-close)
   ;; Give the process buffer the current values of some vars; see
   ;; <https://github.com/purcell/envrc/issues/22>.
-  (cl-letf* (((default-value 'process-environment) process-environment)
-             ((default-value 'exec-path)           exec-path))
-    (make-process
-     :name            racket--cmd-process-name
-     :connection-type 'pipe
-     :noquery         t
-     :coding          'utf-8
-     :buffer          (get-buffer-create (concat " *" racket--cmd-process-name "*"))
-     :stderr          (make-pipe-process
-                       :name     (concat racket--cmd-process-name "-stderr")
-                       :buffer   nil
-                       :noquery  t
-                       :coding   'utf-8
-                       :filter   #'racket--cmd-process-stderr-filter
-                       :sentinel #'ignore)
-     :command         (list racket-program
-                            (funcall racket-adjust-run-rkt racket--run.rkt)
-                            "--auth"
-                            (setq racket--cmd-auth (format "token-%x" (random)))
-                            (if (and (boundp 'image-types)
-                                     (fboundp 'image-type-available-p)
-                                     (or (and (memq 'svg image-types)
-                                              (image-type-available-p 'svg))
-                                         (and (memq 'imagemagick image-types)
-                                              (image-type-available-p 'imagemagick))))
+  (let ((cmd (list racket-program
```

Comment:
This assumes that `racket-program` will work on both local and remote -- e.g. that `racket` is good on both, as opposed to `/path/to/specific/version/of/racket` on one but not the other. That assumption will probably be fine for most people, so maybe it's fine to punt on this, as opposed to adding Yet Another Customization Variable? What do you think?